packages/block-editor/src/components/inserter/index.js

```diff
@@ -1,29 +1,45 @@
+/**
+ * External dependencies
+ */
+import {
+	get,
+} from 'lodash';
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { __, _x, sprintf } from '@wordpress/i18n';
 import { Dropdown, IconButton } from '@wordpress/components';
 import { Component } from '@wordpress/element';
-import { withSelect } from '@wordpress/data';
+import { withDispatch, withSelect } from '@wordpress/data';
 import { compose, ifCondition } from '@wordpress/compose';
+import {
+	createBlock,
+	getBlockType,
+} from '@wordpress/blocks';
 
 /**
  * Internal dependencies
  */
 import InserterMenu from './menu';
 
-const defaultRenderToggle = ( { onToggle, disabled, isOpen } ) => (
-	<IconButton
-		icon="insert"
-		label={ __( 'Add block' ) }
-		labelPosition="bottom"
-		onClick={ onToggle }
-		className="editor-inserter__toggle block-editor-inserter__toggle"
-		aria-haspopup="true"
-		aria-expanded={ isOpen }
-		disabled={ disabled }
-	/>
-);
+const defaultRenderToggle = ( { onToggle, disabled, isOpen, blockTitle } ) => {
+	let label = sprintf( _x( 'Add %s', 'directly add the allowed block' ), blockTitle );
+	if ( blockTitle === '' ) {
+		label = 'Add block';
+	}
```

Comment:
I think I'd avoid the pattern of assigning `label` once and then again after looking at `blockTitle`. Instead, I'd define `label` with a `const` and a ternary. Just my suggestion, because it feels awkward to call `_x` in vain and `sprintf` with a potentially missing value.